sql/base/characters_database.sql

```diff
@@ -1536,15 +1632,21 @@ CREATE TABLE `gm_ticket` (
   `posY` float NOT NULL DEFAULT '0',
   `posZ` float NOT NULL DEFAULT '0',
   `lastModifiedTime` int(10) unsigned NOT NULL DEFAULT '0',
-  `closedBy` int(10) NOT NULL DEFAULT '0',
-  `assignedTo` int(10) unsigned NOT NULL DEFAULT '0' COMMENT 'GUID of admin to whom ticket is assigned',
+  `closedBy` int(10) unsigned DEFAULT NULL,
+  `assignedTo` int(10) unsigned DEFAULT NULL,
   `comment` text NOT NULL,
   `response` text NOT NULL,
   `completed` tinyint(3) unsigned NOT NULL DEFAULT '0',
   `escalated` tinyint(3) unsigned NOT NULL DEFAULT '0',
   `viewed` tinyint(3) unsigned NOT NULL DEFAULT '0',
   `needMoreHelp` tinyint(3) unsigned NOT NULL DEFAULT '0',
-  PRIMARY KEY (`id`)
+  PRIMARY KEY (`id`),
+  KEY `playerGuid` (`playerGuid`),
+  KEY `closedBy` (`closedBy`),
+  KEY `assignedTo` (`assignedTo`),
+  CONSTRAINT `gm_ticket_ibfk_1` FOREIGN KEY (`playerGuid`) REFERENCES `characters` (`guid`) ON DELETE CASCADE ON UPDATE CASCADE,
+  CONSTRAINT `gm_ticket_ibfk_2` FOREIGN KEY (`closedBy`) REFERENCES `characters` (`guid`) ON DELETE CASCADE ON UPDATE CASCADE,
```

Comment:
It doesn't make sense to delete the ticket once the one closing it is deleted. It rather makes more sense if it was nulled instead, seeing the creator itself (which is the only reference that actually matters) may still exist.